H7A3Zi_AzureRTOS_ThreadX/Core/Src/app_threadx.c

```diff
@@ -66,7 +78,11 @@ UINT App_ThreadX_Init(VOID *memory_ptr)
   /* USER CODE END App_ThreadX_MEM_POOL */
 
   /* USER CODE BEGIN App_ThreadX_Init */
-
+  status = tx_semaphore_create(&semaphore, "semaphore-1", 1);
```

Comment:
Is it binary semaphore or counting ?

Reply:
Binary Semaphore, as we are using one Semaphore only with the initial count of 1.